File: lemur-app/src/main/java/io/github/malczuuu/lemur/app/core/player/PlayerCommand.java
package io.github.malczuuu.lemur.app.core.player;

import io.github.malczuuu.lemur.app.core.Command;

public interface PlayerCommand extends Command {

  record CreatePlayer(String name) {}

  record UpdatePlayer(String id, String name, Long version) {}

  record BanPlayer(String id) {}

  record UnbanPlayer(String id) {}
}

File: lemur-app/src/main/kotlin/io/github/malczuuu/lemur/app/core/player/PlayerCommand.kt
package io.github.malczuuu.lemur.app.core.player

import io.github.malczuuu.lemur.app.core.Command

interface PlayerCommand : Command {

    data class CreatePlayer(val name: String) : PlayerCommand

    data class UpdatePlayer(val id: String, val name: String, val version: Long) : PlayerCommand

    data class BanPlayer(val id: String) : PlayerCommand

    data class UnbanPlayer(val id: String) : PlayerCommand
}
